app/controllers/api/v1/import_controller.rb

```diff
@@ -0,0 +1,137 @@
+  private
+
+  # Set HTTP responses based on success and error flags
+  def response_code(messages)
+    if @some_success && @some_errors
+      status = :multi_status
+      messages << "At least one work was not imported. Please check the works array for further information."
+    elsif !@some_success && @some_errors
+      status = :unprocessable_entity
+      messages << "None of the works were imported. Please check the works array for further information."
+    else
+      status = :created
+      messages << "All works were successfully imported."
+    end
+    [status, messages]
+  end
+
+  # Use the story parser to import works from the chapter URLs,
+  # and set success or error flag depending on the outcome
+  # Returns a hash
+  def import_work(archivist, external_work)
+    work_status, work_messages = work_errors(external_work)
+    work_url = ""
+    original_url = []
+    if work_status == :ok
+      urls = external_work[:chapter_urls]
+      original_url = urls.first
+      storyparser = StoryParser.new
+      options = options(archivist, external_work)
+      begin
+        work = storyparser.download_and_parse_chapters_into_story(urls, options)
+        work.save
+        @some_success = true
+        work_status = :created
+        work_url = work_url(work)
+        work_messages << 'Successfully created work "' + work.title + '"'
+      rescue => exception
+        @some_errors = true
+        work_status = :unprocessable_entity
+        work_messages << exception.message
+      end
+    end
+
+    {status: work_status,
+     url: work_url,
+     original_url: original_url,
+     messages: work_messages}
+  end
+
+  # Top-level error handling: returns a 403 forbidden if a valid archivist isn't supplied and a 400
+  # if no works are supplied. If there is neither a valid archivist nor valid works, a 400 is returned
+  # ith both errors as a message
+  def batch_errors(archivist, external_works)
+    status = :bad_request
+    errors = []
+
+    unless archivist && archivist.is_archivist?
+      status = :forbidden
+      errors << "Only an Archive archivist can import works through the API."
+    end
+
+    if external_works.nil? || external_works.empty?
+      errors << "No work URLs were provided."
+    end
+    status = :ok if errors.empty?
+    [status, errors]
+  end
+
+  # Work-level error handling for requests that are incomplete or too large
+  def work_errors(work)
+    status = :bad_request
+    errors = []
+    urls = work[:chapter_urls]
+    if urls.nil?
+      errors << "This work doesn't contain chapter_urls. Works can only be imported from publicly-accessible URLs."
+    elsif urls.length >= ArchiveConfig.IMPORT_MAX_WORKS_BY_ARCHIVIST
+      errors << "Some works contain too many external links. The Archive limit is #{ ArchiveConfig.IMPORT_MAX_WORKS_BY_ARCHIVIST }"
+    end
+    status = :ok if errors.empty?
+    [status, errors]
+  end
+
+  def options(archivist, params)
+    {
+        archivist: archivist,

```

Comment:
Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.